hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssEncryptionKeyHandler.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.tss.handlers;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.state.common.EntityNumber;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.app.spi.workflows.PreCheckException;
+import com.hedera.node.app.spi.workflows.PreHandleContext;
+import com.hedera.node.app.spi.workflows.TransactionHandler;
+import com.hedera.node.app.tss.stores.WritableTssStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Handles the TSS encryption key transaction.
+ */
+@Singleton
+public class TssEncryptionKeyHandler implements TransactionHandler {
+
+    /**
+     * Constructs a {@link TssEncryptionKeyHandler}.
+     */
+    @Inject
+    public TssEncryptionKeyHandler() {}
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void preHandle(@NonNull PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handle(@NonNull final HandleContext context) throws HandleException {
+        requireNonNull(context);
+        final var op = context.body().tssEncryptionKeyOrThrow();
+        final var tssStore = context.storeFactory().writableStore(WritableTssStore.class);
+        final var nodeEntityNumber =
+                EntityNumber.newBuilder().number(context.creatorInfo().nodeId()).build();
+        tssStore.put(nodeEntityNumber, op);
```

Comment:
Instead of storing the entire transaction body, could we just store the `bytes` of the encryption key (which is the only field in the body)?
That would reduce state size and also make the block stream a little smaller.

Reply:
I'm following the conversations inside of [Protobufs for TSS Startup PR](https://github.com/hashgraph/hedera-services/pull/16822#discussion_r1860973876) as they are essentially the same concern as here. Other than that, the other comments are addressed

Reply:
Same as [here](https://github.com/hashgraph/hedera-services/pull/16772#discussion_r1866143719)